hardware_interface/src/component_parser.cpp

```diff
@@ -139,47 +141,29 @@ std::unordered_map<std::string, std::string> parse_parameters_from_xml(
  * \return std::vector< std::__cxx11::string > list of interface types
  * \throws std::runtime_error if the interfaceType text not set in a tag
  */
-std::vector<components::InterfaceInfo> parse_interfaces_from_xml(
-  const tinyxml2::XMLElement * interfaces_it, const char * interfaceTag)
+components::InterfaceInfo parse_interfaces_from_xml(
+  const tinyxml2::XMLElement * interfaces_it)
 {
-  std::vector<components::InterfaceInfo> interfaces;
-
-  while (interfaces_it) {
-    hardware_interface::components::InterfaceInfo interface;
-
-    // Joint interfaces have a name attribute
-    if (std::string(interfaceTag) == "commandInterfaceType") {
-      const std::string interface_name = get_attribute_value(
-        interfaces_it, "name",
-        std::string(interfaceTag));
-      interface.name = interface_name;
-
-      // Optional min/max attributes
-      std::unordered_map<std::string, std::string> interface_params =
-        parse_parameters_from_xml(interfaces_it->FirstChildElement(kParamTag));
-      std::unordered_map<std::string, std::string>::const_iterator interface_param =
-        interface_params.find(kMinTag);
-      if (interface_param != interface_params.end()) {
-        interface.min = interface_param->second;
-      }
-      interface_param = interface_params.find(kMaxTag);
-      if (interface_param != interface_params.end()) {
-        interface.max = interface_param->second;
-      }
-    }
-    // State interfaces have an element to define the type, not a name attribute
-    if (std::string(interfaceTag) == "stateInterfaceType") {
-      const std::string interface_type = get_text_for_element(
-        interfaces_it,
-        std::string(interfaceTag) + " type ");
-      interface.name = interface_type;
-    }
-
-    interfaces.push_back(interface);
-    interfaces_it = interfaces_it->NextSiblingElement(interfaceTag);
+  hardware_interface::components::InterfaceInfo interface;
+
+  const std::string interface_name = get_attribute_value(
+    interfaces_it, kNameAttribute, interfaces_it->Name());
+  interface.name = interface_name;
+
+  // Optional min/max attributes
+  std::unordered_map<std::string, std::string> interface_params =
+    parse_parameters_from_xml(interfaces_it->FirstChildElement(kParamTag));
+  std::unordered_map<std::string, std::string>::const_iterator interface_param =
```

Comment:
```suggestion
  auto interface_param =
```
auto's payback to long & ugly iterator names

Reply:
that's been like this on master already. Happy to address this though.